pkg/kv/store_message.go

```diff
@@ -64,8 +64,66 @@ func (s *StoreMessage) Delete(ctx context.Context, path string) error {
 	return err
 }
 
-// TODO(niro): implement Scan when required
+// Scan returns a prefix iterator which returns entries in deserialized format. Scan in store message implementation is msg specific
+// and therefore must be PREFIX bound
+func (s *StoreMessage) Scan(ctx context.Context, msg protoreflect.Message, prefix string) (*MessageIterator, error) {
+	return NewMessageIterator(ctx, s.Store, msg, prefix)
+}
 
 func (s *StoreMessage) Close() {
 	s.Store.Close()
 }
+
+type MessageIterator struct {
+	itr EntriesIterator
+	msg protoreflect.Message
+	err error
+}
+
+type MessageEntry struct {
+	Key   string
+	Value protoreflect.ProtoMessage
+}
+
+func NewMessageIterator(ctx context.Context, store Store, msg protoreflect.Message, prefix string) (*MessageIterator, error) {
+	itr, err := ScanPrefix(ctx, store, []byte(prefix))
+	if err != nil {
+		return nil, fmt.Errorf("create prefix iterator: %w", err)
+	}
+	return &MessageIterator{itr: itr, msg: msg}, nil
+}
+
+func (m *MessageIterator) Next() bool {
+	if m.err != nil {
+		return false
+	}
+	return m.itr.Next()
+}
+
+func (m *MessageIterator) Entry() *MessageEntry {
+	entry := m.itr.Entry()
+	if entry == nil {
```

Comment:
What if there's an error in the underlying iterator? I think the `Err()` functionality is fine, but not sure about the `Next()` in such a case.